### ISSUE TITLE
I have one question about CrossMixture and HomogeneousNoMixture settings

### ISSUE THREAD

**Issue by Katono5** [LINs-lab/ttab#20]
Hi, 
Based on my understanding of these two settings, HomogeneousNoMixture will pass test samples within one domain and the other without mixing them together, but CrossMixture will first merge two datasets then pass the mixed test samples to model, so the data batches send to model is largely different, but when I print the logits out from one_adapt_step function, 
https://github.com/LINs-lab/ttab/blob/815fc942f9630f2499ad2e0dd556fc199222e4da/ttab/model_adaptation/tent.py#L72
https://github.com/LINs-lab/ttab/blob/815fc942f9630f2499ad2e0dd556fc199222e4da/ttab/model_adaptation/tent.py#L83
I found out that no matter I use CrossMixture or HomogeneousNoMixture setting, the output is exactly same, so the input is same as well. Which is really weird to me since I suppose the input would be really different. 
But based on the final result the acc after adaptation is indeed different between these two settings so the setting indeed works, I pasted two print_out result below so that you can understand my question.

Could you help me find out what's the reason for the same output?

![image](https://github.com/LINs-lab/ttab/assets/61302913/4d00b71b-757b-4e57-98c4-2850adf56785)
![image](https://github.com/LINs-lab/ttab/assets/61302913/a57b8712-9d08-4659-9f20-3d0ceaa52369)

**Comment by Katono5** [comment]
Further information about my problem, here softmax-ed output at the very last step after adaptation using HomogeneousNoMixture and CrossMixture.
![image](https://github.com/LINs-lab/ttab/assets/61302913/e05e4a1e-5831-4a98-ad40-cd72325502bd)
![image](https://github.com/LINs-lab/ttab/assets/61302913/86eed15c-2d2f-43b8-8403-8c0ca46f8497)

**Comment by MarcellusZhao** [comment]
Hi,

To quickly check this issue, I output the indices of merged dataset in both cases (i.e., `HomogeneousNoMixture` and `CrossMixture`). And it seems to me both settings here work correctly. I show the the indices as follows,

![CleanShot 2024-04-06 at 15 39 02@2x](https://github.com/LINs-lab/ttab/assets/50657356/7663b19f-7e68-4816-8594-6d4e88a31c85)
![CleanShot 2024-04-06 at 15 40 25@2x](https://github.com/LINs-lab/ttab/assets/50657356/dc2a7e01-f237-4963-8f33-dd372b8ddef1)

**Comment by QiuWilliam** [comment]
Hii,if I want to test the adaptation method in the case of multi-domain mixture (where a batch contains data from multiple domains), the results for CIFAR10-C,what should I do to the cifar10c.py.I have choose the CrossMixture,but the results still shows the test domain is only one test set of CIFAR10-C.The test domain is separete.What should I do to combine the whole CIFAR10C test set.
![捕获](https://github.com/LINs-lab/ttab/assets/146973835/b42961f0-9d17-41a4-8cfe-8fe1cbe5eac8)
![image](https://github.com/LINs-lab/ttab/assets/146973835/3957cf17-b530-4ba1-a846-d7c896b47d25)
